### PR TITLE
Major cleanup

### PR DESCRIPTION
**Pull request by wikman**
Fixed some bugs, de-duplicated lots of code, improve reproducibility.

There are some things I think should be changed but left in the code because they would deprecate the trained models. For example, the reconstruction loss should probably use element-wise cross-entropy and the log variance should not be forced up with softplus.

It would also be preferable to use `nn.Embedding` instead of the one-hot encoding and preserve the structure of the sequences throughout the network to allow for other encoder and decoder architectures.

### CODE REVIEW

**Comment by limitloss** [comment]
Thanks for the PL, I appreciate the work that's been put in. I've had a think about it and had some discussions, and I think its best if this remains in its own separate fork.
